hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/protocol/dmr/StatisticsControl.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2015-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.agent.monitor.protocol.dmr;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.hawkular.agent.monitor.log.AgentLoggers;
+import org.hawkular.agent.monitor.log.MsgLogger;
+import org.hawkular.agent.monitor.util.WildflyCompatibilityUtils;
+import org.hawkular.dmr.api.OperationBuilder;
+import org.hawkular.dmr.api.OperationBuilder.CompositeOperationBuilder;
+import org.hawkular.dmr.api.OperationBuilder.OperationResult;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+
+/**
+ * Turns on or off statistics for several WildFly subsystems.
+ */
+public class StatisticsControl {
+    private static final MsgLogger log = AgentLoggers.getLogger(StatisticsControl.class);
+
+    public StatisticsControl() {
+    }
+
+    public void enableStatistics(ModelControllerClient mcc) {
+        execute(mcc, true);
+    }
+
+    public void disableStatistics(ModelControllerClient mcc) {
+        execute(mcc, false);
+    }
+
+    private void execute(ModelControllerClient mcc, boolean enable) {
+
+        try {
+            List<String> hosts = getDomainHosts(mcc);
+            if (hosts.isEmpty()) {
+                executeForServer(mcc, enable, "");
+            } else {
+                for (String host : hosts) {
+                    List<String> servers = getDomainHostServers(mcc, host);
+                    for (String server : servers) {
+                        // I was trying to be too clever. This would work awesomely if WildFly would just allow it.
+                        // However, this error occurs when you try to set attributes on slave servers:
+                        // "User operations are not permitted to directly update the persistent configuration
+                        // of a server in a managed domain."
+                        // So, for now, comment this out until perhaps in the future WildFly/EAP will allow this.
+                        // This means when in domain mode, the user must manually turn on statistics themselves.
```

Comment:
Wouldn't the slaves be enabled if you enable the controller?

Reply:
no, the slaves configuration here are not affected by changes in the host controller (in fact, for example, there is no /subsystem=ejb3 in the host controller anyway.